### ISSUE TITLE
Counterintuitive functionality when viewing mapped TIs from Grid View

### ISSUE THREAD

**Issue by RNHTTR** [apache/airflow#34489]
### Apache Airflow version

2.7.1

### What happened

When a user clicks on a mapped TI, the "[] Mapped Tasks" button is replaced by a "Logs" button, and returning to the "Mapped task summary" moves to the side of the view:

<img width="856" alt="image" src="https://github.com/apache/airflow/assets/25823361/413b3399-97ee-4b44-8af4-c3f93f498008">

___
___

<img width="868" alt="image" src="https://github.com/apache/airflow/assets/25823361/c3f7b945-6b1a-4470-8428-949746b4e035">

### What you think should happen instead

Instead, I think the "[] Mapped Tasks" button should remain in the same place, and instead, a "Logs [0]" tab/button should be added just next to the "[] Mapped Tasks" button. In this example, "[0]" represents  that 0th mapped TI has been clicked into:

<img width="866" alt="image" src="https://github.com/apache/airflow/assets/25823361/183ae3bd-587c-45d1-b868-67a7339c1623">

### How to reproduce

Open the UI in 2.7 for a mapped task

### Operating System

n/a

### Versions of Apache Airflow Providers

n/a

### Deployment

Astronomer

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by bbovenzi** [comment]
Makes sense. That "Back to dynamic tasks" always felt a bit hacky.

**Comment by hainenber** [comment]
Hi there, I could work on this. Pls help assigning. Thank you

**Comment by bbovenzi** [comment]
Assigned! Reach out on slack if you want any help

**Comment by lotrias17** [comment]
Hello, could I get assigned to this issue? Thank you.

**Comment by potiuk** [comment]
Sure. But check with @hainenber - maybe they still want to work on it :)

**Comment by hainenber** [comment]
Hi there, the task is free to pick up. Sorry, I'm not skillful for this one 😮‍💨

**Comment by lotrias17** [comment]
I'm very sorry I had mistaken the issue for another one. I am also not skillfull enough for this one. I wanted to take issue #35946. Could you please unassign me? Once again I am very sorry for the misunderstanding.

**Comment by RNHTTR** [comment]
not stale

**Comment by bbovenzi** [comment]
Yes, we should make it easier to switch between mapped tasks in the Airflow 3 UI. What is we changed the map Index value in the header card to be a dropdown to quickly change?

<img width="525" alt="Image" src="https://github.com/user-attachments/assets/691c65f7-8004-442b-b62a-a3fd3a5ded85" />

**Comment by bbovenzi** [comment]
What about this?

<img width="745" alt="Image" src="https://github.com/user-attachments/assets/69a7e8c4-0b49-40f2-b156-f0d12ac56577" />

**Comment by RNHTTR** [comment]
Looks great imo
